pin-project-internal/src/pin_project/mod.rs

```diff
@@ -228,6 +228,10 @@ impl Context {
                 // '__UnpinStruct' type must also be public. However, we take
                 // steps to ensure that the user can never actually reference
                 // this 'public' type. These steps are described below
+                /// A struct generated by pin-project to provide an appropriate
+                /// `Unpin` implementation, this type's `Unpin` implementation
+                /// uses exactly the same conditions as the original type's
+                /// `Unpin` implementation.
```

Comment:
I believe adding this will improve the generated documentation a little.

<h2 id='structs' class='section-header'><a href="#structs">Structs</a></h2>
<table><tr class='module-item'><td><a class="struct" href="struct.PublicStruct.html" title='futures_async_stream::PublicStruct struct'>PublicStruct</a></td><td class='docblock-short'></td></tr><tr class='module-item'><td><a class="struct" href="struct.__UnpinStructPublicStruct.html" title='futures_async_stream::__UnpinStructPublicStruct struct'>__UnpinStructPublicStruct</a></td><td class='docblock-short'><p>A struct generated by pin-project to provide an appropriate
<code>Unpin</code> implementation, this type's <code>Unpin</code> implementation
uses exactly the same conditions as the original type's
<code>Unpin</code> implementation.</p>
</td></tr></table>

Reply:
(Perhaps we can improve the documentation and naming more.)

Reply:
@Aaron1011 @Nemo157 Any thoughts?